core/src/telemetry/mod.rs

```diff
@@ -226,19 +258,10 @@ pub fn telemetry_init(opts: &TelemetryOptions) -> Result<&'static GlobalTelemDat
             if let Some(ref logger) = opts.logging {
                 match logger {
                     Logger::Console => {
-                        // TODO: this is duplicated below and is quite ugly, remove the duplication
                         if opts.tracing.is_none() {
-                            let pretty_fmt = tracing_subscriber::fmt::format()
-                                .pretty()
-                                .with_source_location(false);
-                            let reg = tracing_subscriber::registry()
-                                .with((opts).try_get_env_filter()?)
-                                .with(
-                                    tracing_subscriber::fmt::layer()
-                                        .with_target(false)
-                                        .event_format(pretty_fmt),
-                                );
-                            tracing::subscriber::set_global_default(reg)?;
+                            let reg =
+                                tracing_subscriber::registry().with(opts.try_get_env_filter()?);
+                            make_console_logger!(opts, reg);
```

Comment:
This is really just for me running my own tests. Besides, configuring logging with env vars *already* is possible (`RUST_LOG`) and indeed generally quite common, not sure what the vendetta is here but this is far from an unusual thing to do.

Reply:
You can use custom setup code in tests.
I'd really prefer leaving our libraries as unopinionated as possible wrt configuration management.
We've already stuck to this throughout the API, why make an exception now?

Reply:
I don't mind env vars if they are neither required nor clash. If those two requirements are satisfied, they might as well not exist from lang POV (I'll never document them for user use).

Reply:
Because I have no intention of exposing this to users. Chad is right, this effectively doesn't exist and has a unique name. Reading it and setting it from my tests is actually a fairly substantial lift to make work for every kind of UT. Many UT's don't even start a worker, and thus there's no configuration to modify. The setting in this case needs to be way down deep.